crates/interpreter/src/exec.rs

```diff
@@ -1066,9 +1060,9 @@ impl<'m> Thread<'m> {
         self.values().drain(values_len - values_cnt .. values_len - arity);
         self.label().values_cnt += arity;
         match kind {
-            LabelKind::Loop(state) => unsafe {
-                self.frame().side_table = state.side_table;
-                self.parser.restore(state.parser_data)
+            LabelKind::Loop(parser_data) => unsafe {
+                self.take_jump(offset);
+                self.parser.restore(parser_data)
```

Comment:
This line shouldn't be needed. And we can also remove the `saved` from `LabelKind::Loop`.